// perfecto-cypress-sdk/lib/init.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _fs = _interopRequireDefault(require("fs"));

var _consts = require("./common/consts");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getPerfectoConfigObject = (cloud, securityToken, testsPath) => ({
  credentials: {
    cloud: cloud || '<REPLACE_THIS_WITH_CLOUD_NAME>',
    securityToken: securityToken || '<REPLACE_THIS_WITH_SECURITY_TOKEN>'
  },
  capabilities: [{
    deviceType: 'Web',
    platformName: 'Windows',
    platformVersion: '10',
    browserName: 'Chrome',
    browserVersion: 'latest',
    resolution: '1024x768',
    location: 'US East'
  }],
  env: {
    CYPRESS_PAGE_LOAD_TIMEOUT: '10000'
  },
  framework: _consts.SUPPORTED_FRAMEWORKS.CYPRESS,
  nodeVersion: '12',
  tests: {
    path: testsPath || '<REPLACE_THIS_WITH_PATH_TO_CYPRESS_FOLDER>',
    artifactKey: '',
    ignore: ['**/screenshots/**', '**/videos/**'],
    specsExt: '**/*.spec.js'
  },
  reporting: {
    job: {
      name: 'cypress_first_job',
      number: 1,
      branch: ''
    },
    project: {
      name: 'My_Cypress_project',
      version: 'v1.0'
    },
    customFields: [],
    author: 'sdet1@awesomecompany.com',
    tags: ['cypress']
  }
});

var _default = (cloud, securityToken, testsRoot) => {
  var perfectoConfigPath = 'perfecto-config.json';
  var perfectoConfigObject = getPerfectoConfigObject(cloud, securityToken, testsRoot);

  _fs.default.writeFileSync(perfectoConfigPath, JSON.stringify(perfectoConfigObject, null, 2));

  console.warn('Make sure to import perfecto-cypress-reporter -> https://www.npmjs.com/package/perfecto-cypress-reporter');
};

exports.default = _default;
