// perfecto-cypress-sdk/lib/common/consts.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.globalIgnorePatterns = exports.REPOSITORY_FOLDER_TYPES = exports.StatusIcons = exports.ExecutionState = exports.SessionState = exports.ResultState = exports.TestResults = exports.SUPPORTED_FRAMEWORKS = void 0;
var SUPPORTED_FRAMEWORKS = {
  CYPRESS: 'CYPRESS'
};
exports.SUPPORTED_FRAMEWORKS = SUPPORTED_FRAMEWORKS;
var TestResults = {
  PASSED: 'PASSED',
  FAILED: 'FAILED'
};
exports.TestResults = TestResults;
var ResultState = {
  SUCCESS: 'SUCCESS',
  FAILED: 'FAILED',
  BLOCKED: 'BLOCKED',
  ABORTED: 'ABORTED'
};
exports.ResultState = ResultState;
var SessionState = {
  INITIALIZING: 'INITIALIZING',
  EXECUTING: 'EXECUTING',
  DONE: 'DONE'
};
exports.SessionState = SessionState;
var ExecutionState = {
  ALLOCATING: 'ALLOCATING',
  INITIALIZING: 'INITIALIZING',
  EXECUTING: 'EXECUTING',
  POST_EXECUTING: 'POST_EXECUTING',
  DONE: 'DONE'
};
exports.ExecutionState = ExecutionState;
var StatusIcons = {
  PASSED: '✔︎',
  FAILED: '✖'
};
exports.StatusIcons = StatusIcons;
var REPOSITORY_FOLDER_TYPES = ['PRIVATE', 'PUBLIC', 'GROUP'];
exports.REPOSITORY_FOLDER_TYPES = REPOSITORY_FOLDER_TYPES;
var globalIgnorePatterns = ['**/node_modules/**', '**/perfecto-cypress.zip'];
exports.globalIgnorePatterns = globalIgnorePatterns;
